### PR TITLE
Outcrop improvements

### PR DESCRIPTION
**Pull request by lstein**
This is a re-do of PR #1414 with missed comments addressed.

### CODE REVIEW

File: ldm/generate.py [hunk 1]
```diff
@@ -561,17 +561,22 @@ def apply_postprocessor(
...
         seed   = None
-        image_metadata = None
         prompt = None
 
         args   = metadata_from_png(image_path)
-        seed   = args.seed
-        prompt = args.prompt
-        print(f'>> retrieved seed {seed} and prompt "{prompt}" from {image_path}')
+        if opt.seed is not None:
+            seed = opt.seed
```
Comment (Kyle0654) [inline ldm/generate.py:568]: What if `opt.seed` is negative?

File: scripts/invoke.py [hunk 2]
```diff
@@ -614,10 +617,16 @@ def do_textmask(gen, opt, callback):
     )
 
 def do_postprocess (gen, opt, callback):
-    file_path = opt.prompt     # treat the prompt as the file pathname
+    file_path = opt.prompt      # treat the prompt as the file pathname
+    if opt.new_prompt is not None:
+        opt.prompt = opt.new_prompt
+    else:
+        opt.prompt = None
```
Comment (Kyle0654) [inline scripts/invoke.py:624]: Should this be clearing the prompt?

**Review by Kyle0654 — APPROVED** [review]
I added a couple of questions, but I'm not sure if they're issues or not.

File: ldm/generate.py (hunk 1 above)
Comment (lstein) [reply ldm/generate.py:568]: Well, this gets fixed in the generator, which also looks for negative seeds, but that's confusing.
Replaced with more readable:
```
        args = metadata_from_png(image_path)
        seed = opt.seed or args.seed
        if seed is None:
            seed = random.randrange(0, np.iinfo(np.uint32).max)
        elif seed <= 0:
            seed = random.randrange(0, np.iinfo(np.uint32).max)
```

**Pull request merged**